lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -6,13 +6,6 @@ module ActiveRecordHostPool
   module DatabaseSwitch
     def self.included(base)
       base.class_eval do
-        attr_reader(:_host_pool_current_database)
-
-        def _host_pool_current_database=(database)
-          @_host_pool_current_database = database
-          @config[:database] = _host_pool_current_database if ActiveRecord::VERSION::MAJOR >= 5
-        end
-
```

Comment:
I moved `_host_pool_current_database` and `_host_pool_current_database=` methods down as methods on the module rather than added on through `class_eval`.